build_tools/provision.py

```diff
@@ -0,0 +1,206 @@
+    log(f"Creating directory {output_dir_path}")
+    if os.path.isdir(output_dir_path):
+        log(
+            f"Directory {output_dir_path} already exists, removing existing directory and files"
+        )
+        shutil.rmtree(output_dir_path)
+    os.mkdir(output_dir_path)
+    log(f"Created directory {output_dir_path}")
+
+
+def _get_github_release_assets(release_id, amdgpu_family):
+    """
+    Makes an API call to retrieve the release's assets, then retrieves the asset matching the amdgpu family
+    """
+    github_release_url = (
+        f"https://api.github.com/repos/ROCm/TheRock/releases/tags/{release_id}"
+    )
+    headers = {
+        "Accept": "application/vnd.github+json",
+        "X-GitHub-Api-Version": "2022-11-28",
+    }
+    response = requests.get(github_release_url, headers=headers)
+    if response.status_code == 403:
+        log(
+            f"Error when retrieving GitHub release assets for release ID {release_id}. This is most likely a rate limiting issue, so please try again"
+        )
+        return
+    elif response.status_code != 200:
+        log(
+            f"Error when retrieving GitHub release assets for release ID {release_id}. Exiting..."
+        )
+        return
+
+    release_data = response.json()
+
+    # We retrieve the most recent release asset that matches the amdgpu_family
+    # In the cases of "nightly-release" or "dev-release", this will retrieve the most recent release asset
+    asset_data = sorted(
+        release_data["assets"], key=lambda item: item["updated_at"], reverse=True
+    )
+    for asset in asset_data:
+        if amdgpu_family in asset["name"]:
+            return asset
+    return None
+
+
+def _download_github_release_asset(asset_data, output_dir):
+    """
+    With the GitHub asset data, this function downloads the asset to the output_dir
+    """
+    asset_name = asset_data["name"]
+    asset_url = asset_data["url"]
+    destination = Path(output_dir) / asset_name
+    headers = {"Accept": "application/octet-stream"}
+    # Making the API call to retrieve the asset
+    response = requests.get(asset_url, stream=True, headers=headers)
+
+    # Downloading the asset in chunks to destination
+    # In order to get better visibility on downloading status, tqdm adds a progress bar
+    total_size = int(response.headers.get("content-length", 0))
+    block_size = 1024
+    with tqdm(total=total_size, unit="B", unit_scale=True) as progress_bar:
+        with open(destination, "wb") as file:
+            for chunk in response.iter_content(block_size * block_size):
+                progress_bar.update(len(chunk))
+                file.write(chunk)
+
+    # After downloading the asset, untar-ing the file
+    _untar_files(output_dir, destination)
+
+
+def retrieve_artifacts_by_ci(args):
+    """
+    If the user requested TheRock artifacts by CI (runner ID), this function will retrieve those assets
+    """
+    runner_id = args.runner_id
```

Comment:
```suggestion
    run_id = args.run_id
```
Wdyt about reusing the variables as they are using within the workflows here?